ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -13477,15 +13477,28 @@ ASTNode analyzeCreateTable(
       }
     }
 
-    if (partitionTransformSpecExists) {
-      try {
-        HiveStorageHandler storageHandler = HiveUtils.getStorageHandler(conf, storageFormat.getStorageHandler());
-        if (!storageHandler.supportsPartitionTransform()) {
-          throw new SemanticException("Partition transform is not supported for " +
-              storageHandler.getClass().getName());
+    HiveStorageHandler handler;
+    try {
+      handler = HiveUtils.getStorageHandler(conf, storageFormat.getStorageHandler());
```

Comment:
Could it be that the `storageFormat.getStorageHandler()` is null? Like for native tables?

Reply:
Yes, the storage handler can be null in the case of native tables, but this is handled inside of `HiveUtils.getStorageHandler()`